### PR TITLE
Synchronise data model with v1 specification.

### PR DESCRIPTION
**Pull request by fasaxc**
* Move data model definitions to shared file.
* Add v1 prefix to keys.
* Add missing workload_id to endpoint keys WARNING: this is just using the endpoint ID because I didn't know what it should be!

### CODE REVIEW

**Comment by fasaxc** [comment]
This needs to go into the release.  I think the most important gate for merging this in is running the FVs against it.

**Comment by fasaxc** [comment]
Discussed with the Maine Coon team about calico-docker.  Turns out that the changes to python-etcd that went in earlier make updating calico-docker much more painful than I'd thought.  They've taken on the action to do that work next time they rev the version of felix they're pulling into their build.  calico-docker compatibility shouldn't block this PR from going in.

**Comment by fasaxc** [comment]
@neiljerram I think this needs to go into the release but I didn't manage to get a clean FV run against it tonight.  When I run a single FV, it fails very quickly with a 500 Internal Server Error (presumably from the OpenStack node) but I couldn't find anything in neutorn/server.log.

You might want to debug on the calicobobcat-ci01 control server, which is patched with the bobcat-integration branch (which has this change and my other pending changes merged in).  I run the test via this jenkins job: http://calicobobcat-jenkins.gulfstream:8080/job/test-class/ but the diags seem to be pretty poor.  It fails almost instantly.

File: calico/felix/fetcd.py [hunk 1]
```diff
@@ -284,11 +274,11 @@ def watch_etcd(self, update_splitter):
...
                                  "resync.", response.action, response.key)
                     continue_polling = False
-                if response.key.startswith("/calico/config"):
+                if response.key.startswith(CONFIG_DIR):
                     _log.warning("Global config changed but we don't "
-                                 "yet support dynamic config response: %s",
-                                 response)
-                if response.key.startswith(self.my_config_prefix):
+                                 "yet support dynamic config key=%s",
+                                 response.key)
```
Comment (neiljerram) [inline calico/felix/fetcd.py]: WHy the change here from response to response.key ?

File: calico/openstack/t_etcd.py [hunk 2]
```diff
@@ -22,16 +22,19 @@
...
+                                 key_for_config)
 from calico.openstack.transport import CalicoTransport
 
 LOG = None
 OPENSTACK_ENDPOINT_RE = re.compile(
-    r'^/calico/host/(?P<hostname>[^/]+)/.*openstack.*/endpoint/(?P<endpoint_id>[^/]+)')
-OPENSTACK_POLICY_RE = re.compile(
-    r'^/calico/policy/profile/(?P<profile_id>[^/]+)/tags')
+    r'^' + VERSION_DIR +
+    r'/host/(?P<hostname>[^/]+)/.*openstack.*/endpoint/(?P<endpoint_id>[^/]+)')
```
Comment (neiljerram) [inline calico/openstack/t_etcd.py]: Can we use HOST_DIR here instead of VERSION_DIR + /host ?

**Comment by neiljerram** [comment]
Couple of comments above.  I also have a tiny delta to push shortly, with changes that I made myself.

File: calico/felix/fetcd.py (hunk 1 above)
Comment (fasaxc) [inline calico/felix/fetcd.py]: I think that might be a merge conflict resolution mistake.  I changed it to response so that it dumps the whole response including its key, action and data.  response is correct.

File: calico/openstack/t_etcd.py (hunk 2 above)
Comment (fasaxc) [inline calico/openstack/t_etcd.py]: Yes, I'll add that in.

**Comment by fasaxc** [comment]
@neiljerram signed this off for me to self-merge after I made those markups.
